### ISSUE TITLE
Cannot Run HRDetect_pipeline function

### ISSUE THREAD

**Issue by MariaCartolano** [Nik-Zainal-Group/signature.tools.lib#19]
Hi,

I am trying to run the HRDetect_pipeline function but I am confronted with the following problem:

_**[info HRDetect_pipeline] Running Signature fit with 100 bootstraps. Increase sparsity by removing exposures with 5% threshold of total mutations and 0.05 threshold of p-value, i.e. exposure of a signature in a sample is set to zero if the probability of having less than 5% of total mutations assigned to that signature is greather than 0.05.
Error in { : 
  task 1 failed - "unable to load shared object '/Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/libs/Rcpp.so':
  dlopen(/Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/libs/Rcpp.so, 6): Symbol not found: _EXTPTR_PTR
  Referenced from: /Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/libs/Rcpp.so
  Expected in: /Library/Frameworks/R.framework/Resources/lib/libR.dylib
 in /Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/libs/Rcpp.so"**_

I suspect this has to do with some bug in the SignatureFit_withBootstrap() because this function outputs the same error when the option method is set to default. However, when I change it to "NNLS", the function works without complaints.

Here is my sessionInfo() output

 sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] signature.tools.lib_0.0.0.9000

loaded via a namespace (and not attached):
 [1] lattice_0.20-41                             prettyunits_1.1.1                          
 [3] Rsamtools_2.4.0                             ps_1.3.3                                   
 [5] Biostrings_2.56.0                           digest_0.6.25                              
 [7] assertthat_0.2.1                            rprojroot_1.3-2                            
 [9] foreach_1.5.0                               R6_2.4.1                                   
[11] GenomeInfoDb_1.24.2                         nnls_1.4                                   
[13] backports_1.1.8                             stats4_4.0.0                               
[15] ggplot2_3.3.2                               pillar_1.4.6                               
[17] zlibbioc_1.34.0                             rlang_0.4.7                                
[19] curl_4.3                                    rstudioapi_0.11                            
[21] callr_3.4.3                                 S4Vectors_0.26.1                           
[23] Matrix_1.2-18                               BiocParallel_1.22.0                        
[25] RCurl_1.98-1.2                              munsell_0.5.0                              
[27] DelayedArray_0.14.1                         compiler_4.0.0                             
[29] rtracklayer_1.48.0                          pkgconfig_2.0.3                            
[31] BiocGenerics_0.34.0                         pkgbuild_1.1.0                             
[33] tidyselect_1.1.0                            SummarizedExperiment_1.18.2                
[35] tibble_3.0.3                                GenomeInfoDbData_1.2.3                     
[37] doMC_1.3.6                                  IRanges_2.22.2                             
[39] codetools_0.2-16                            matrixStats_0.56.0                         
[41] XML_3.99-0.4                                fansi_0.4.1                                
[43] crayon_1.3.4                                dplyr_1.0.0                                
[45] withr_2.2.0                                 GenomicAlignments_1.24.0                   
[47] bitops_1.0-6                                grid_4.0.0                                 
[49] gtable_0.3.0                                lifecycle_0.2.0                            
[51] magrittr_1.5                                scales_1.1.1                               
[53] cli_2.0.2                                   farver_2.0.3                               
[55] XVector_0.28.0                              remotes_2.1.1                              
[57] doParallel_1.0.15                           ellipsis_0.3.1                             
[59] vctrs_0.3.2                                 generics_0.0.2                             
[61] iterators_1.0.12                            tools_4.0.0                                
[63] BSgenome_1.56.0                             Biobase_2.48.0                             
[65] glue_1.4.1                                  purrr_0.3.4                                
[67] processx_3.4.3                              parallel_4.0.0                             
[69] BSgenome.Hsapiens.1000genomes.hs37d5_0.99.1 colorspace_1.4-1                           
[71] BiocManager_1.30.10                         GenomicRanges_1.40.0      

I appreciate any help or idea ,
Maria

**Comment by andreadega** [comment]
Hi Maria,

By default the signature fit function uses the NNLM package, so you may check if you are having problems running that package in particular.

From your error, it appears that a library is missing, so it may be related to some changes in R4.0.0, still likely an issue with NNLM than our code.

Others had some problem installing and running NNLM, you can see here:

https://github.com/linxihui/NNLM/issues/17

Let me know how it goes.
Andrea

**Comment by MariaCartolano** [comment]
Hi Andrea,
thank you for your message. I now managed to run the function successfully. To be honest I am not yet sure what was the problem. I can say for sure that it has to do with the R upgrade to the  3.4 version. Before then, I never had problems running the HRDetect_pipeline function.
Just as a reference for others that may run in to the same errors: in order to fix it I had to install from source the following packages: data.table, rtracklayer and open ssl.

Thank you for your help,
Maria
